### PR TITLE
PR Tracking the development of v0.12

### PR DESCRIPTION
**Pull request by chmp**
Wait with merge until ready to release, to allow reacting to bugs in the 0.11 line

### CODE REVIEW

**Comment by v1gnesh** [comment]
It would be educational to read an ELI5 of this refactoring initiative, if/when you get time to write about it :)

**Comment by chmp** [comment]
@v1gnesh a bit delayed, but I wrote a [short note](https://cprohm.de/blog/serde-arrow-refactor/) on the current changes :)

**Comment by chmp** [comment]
@v1gnesh a bit delayed, but I wrote a [short note](https://cprohm.de/blog/serde-arrow-refactor/) on the current changes :)
